contrib/endpoints/include/api_manager/request.h

```diff
@@ -70,6 +70,9 @@ class Request {
   // Adds a header to backend. If the header exists, overwrite its value
   virtual utils::Status AddHeaderToBackend(const std::string &key,
                                            const std::string &value) = 0;
+
+  // Gets all HTTP request headers.
+  virtual void GetHeaders(std::map<std::string, std::string> *headers) {}
```

Comment:
I have provided all the information in the PR description. This cannot be 0. If this is 0, then ESP that will point to this proxy's commit will have issues creating a request object since it does not implement a pure virtual function. Having this empty body will fix that issue.